src/renderers/dom/shared/hooks/ReactDOMUnknownPropertyHook.js

```diff
@@ -76,54 +59,54 @@ if (__DEV__) {
       // Don't check events in this case.
       return true;
     }
-    warnedProperties[name] = true;
-    var lowerCasedName = name.toLowerCase();
-
-    // data-* attributes should be lowercase; suggest the lowercase version
-    var standardName = DOMProperty.isCustomAttribute(lowerCasedName)
-      ? lowerCasedName
-      : DOMProperty.getPossibleStandardName.hasOwnProperty(lowerCasedName)
-          ? DOMProperty.getPossibleStandardName[lowerCasedName]
-          : null;
 
     var registrationName = EventPluginRegistry.possibleRegistrationNames.hasOwnProperty(
       lowerCasedName,
     )
       ? EventPluginRegistry.possibleRegistrationNames[lowerCasedName]
       : null;
 
-    if (standardName != null) {
+    if (registrationName != null) {
       warning(
         false,
-        'Unknown DOM property %s. Did you mean %s?%s',
+        'Unknown event handler property %s. Did you mean `%s`?%s',
         name,
-        standardName,
+        registrationName,
         getStackAddendum(debugID),
       );
       return true;
-    } else if (registrationName != null) {
+    }
+
+    // data-* attributes should be lowercase; suggest the lowercase version
```

Comment:
Great question. It looks like [data attributes are not case sensitive](https://codepen.io/nhunzaker/pen/EvZxme). What is the expected behavior? 

- Should `data-MyAttribute` be written to the DOM? 
- Should we do this validation (lowercase only names) for all custom attributes?